### DISCUSSION TITLE
weblate upgrade

### DISCUSSION THREAD

**Discussion by rbos** [WeblateOrg/weblate#7750] [Q&A] [answered]
Hi,

I have inherited a Weblate 3.9.1 install. The documentation says that I should update to at least 4.0.4 before going to the latest 4.x branch, but it also says it does some database migrations from 4.2 to 4.3 and some other stuff.

Can I safely go straight from 4.0 to the most recent build, or do I need to do any additional intermediate steps?

**Comment by nijel** [comment]
It should work this way (upgrade to 4.0.x and then upgrade to anything in the 4.x series), but there is a bug in the migrations in one of the releases, and you have to do an intermediate step. I'm not sure where exactly it was, but I think an additional step with 4.1 was needed.

**Reply by rbos** [reply]
"There's a landmine somewhere on this path. I think it's just past the first tree but I'm not sure." :)

**Reply by nijel** [reply]
It will just fail to start, so fixing it is just a matter of trying different versions. I've run into that once when upgrading some ancient version...

**Comment by rbos** [comment, marked as answer]
Update: I went 3.9 -> 4.0.4-1 -> 4.1.1-3 -> 4.13.1 (latest as of time of writing), and it appears to have worked fine.

**Reply by rbos** [reply]
Also: it's not clear anywhere in the documentation but you, eg, edit docker-compose.yml, and change "weblate/weblate" to "weblate/weblate:4.0.4-1" and then do "docker compose up".

**Reply by nijel** [reply]
Pull requests to improve the documentation are welcome 😉
